Python/HAT/hypergraph.py

```
@@ -365,75 +473,58 @@ def set_adjacency_tensor(self, AT=None):
+                self._adjacency_tensor = np.zeros((num_nodes,) * self.order, dtype=int)
+                num_edges = self.incidence_matrix.shape[1]
+
+                # Populate adjacency tensor based on incidence matrix
+                for i in range(num_edges):
+                    nodes = np.where(self.incidence_matrix[:, i] == 1)[0]
+                    if len(nodes) == self.order:
+                        # Use multi-dimensional indexing to set entries for the k-node hyperedge
+                        for nodes_indices in permutations(nodes):
+                            self._adjacency_tensor[tuple(nodes_indices)] = 1
+                    else:
+                        warnings.warn(f"Edge {i} does not have exactly {k} nodes; skipping.")
```

Ruff (0.8.0)

No explicit `stacklevel` keyword argument found

(B028)